main.py

```
@@ -0,0 +1,272 @@
+        pending_nonce_map[address] = 0
+
+async def node_loop():
+    logger.info("Starting MiniChain Node with Smart Contracts")
+
+    state = State()
+    chain = Blockchain()
+    mempool = Mempool()
+
+    pending_nonce_map = {}
+
+    def claim_nonce(address):
```

Ruff (0.15.0)

[warning] Missing return type annotation for private function `claim_nonce`

(ANN202)